cmd/cloudFoundryCreateService_test.go

```diff
@@ -0,0 +1,128 @@
+		if error == nil {
+			assert.Equal(t, "cf", execRunner.Calls[0].Exec)
+			assert.Equal(t, []string{"create-service", "testService", "testPlan", "testName"}, execRunner.Calls[0].Params)
+		}
+	})
+	t.Run("Create service: only tags", func(t *testing.T) {
+		config := cloudFoundryCreateServiceOptions{
+			CfAPIEndpoint:         "https://api.endpoint.com",
+			CfOrg:                 "testOrg",
+			CfSpace:               "testSpace",
+			Username:              "testUser",
+			Password:              "testPassword",
+			CfService:             "testService",
+			CfServiceInstanceName: "testName",
+			CfServicePlan:         "testPlan",
+			CfServiceTags:         "testTag, testTag2",
+		}
+		error := runCloudFoundryCreateService(&config, &telemetryData, &execRunner)
+		if error == nil {
+			assert.Equal(t, "cf", execRunner.Calls[1].Exec)
+			assert.Equal(t, []string{"create-service", "testService", "testPlan", "testName", "-t", "testTag, testTag2"}, execRunner.Calls[1].Params)
+		}
+	})
+	t.Run("Create service: only broker", func(t *testing.T) {
+		config := cloudFoundryCreateServiceOptions{
+			CfAPIEndpoint:         "https://api.endpoint.com",
+			CfOrg:                 "testOrg",
+			CfSpace:               "testSpace",
+			Username:              "testUser",
+			Password:              "testPassword",
+			CfService:             "testService",
+			CfServiceInstanceName: "testName",
+			CfServicePlan:         "testPlan",
+			CfServiceBroker:       "testBroker",
+		}
+		error := runCloudFoundryCreateService(&config, &telemetryData, &execRunner)
+		if error == nil {
+			assert.Equal(t, "cf", execRunner.Calls[2].Exec)
+			assert.Equal(t, []string{"create-service", "testService", "testPlan", "testName", "-b", "testBroker"}, execRunner.Calls[2].Params)
+		}
+	})
+	t.Run("Create service: only config", func(t *testing.T) {
+		config := cloudFoundryCreateServiceOptions{
+			CfAPIEndpoint:         "https://api.endpoint.com",
+			CfOrg:                 "testOrg",
+			CfSpace:               "testSpace",
+			Username:              "testUser",
+			Password:              "testPassword",
+			CfService:             "testService",
+			CfServiceInstanceName: "testName",
+			CfServicePlan:         "testPlan",
+			CfCreateServiceConfig: "testConfig.json",
+		}
+		error := runCloudFoundryCreateService(&config, &telemetryData, &execRunner)
+		if error == nil {
+			assert.Equal(t, "cf", execRunner.Calls[3].Exec)
+			assert.Equal(t, []string{"create-service", "testService", "testPlan", "testName", "-c", "testConfig.json"}, execRunner.Calls[3].Params)
+		}
+	})
+
+	t.Run("Create service: failure, no config", func(t *testing.T) {
+		config := cloudFoundryCreateServiceOptions{}
+		error := runCloudFoundryCreateService(&config, &telemetryData, &execRunner)
+		if error == nil {
+			assert.Equal(t, "cf", execRunner.Calls[4].Exec)
+			assert.Equal(t, []string{"create-service", "", "", ""}, execRunner.Calls[4].Params)
```

Comment:
We have empty parameters? Why are they added if they are empty?

Reply:
This part has been removed, we want to assert for an Error here which is now implemented.